### PR TITLE
Feature/enable wizard navigating routes

### PR DESCRIPTION
**Pull request by lilachic**
### Description

Enable using of browser's back and forward buttons. Steps are navigable with router.

### Related issues

This solves the issue https://github.com/CSCfi/metadata-submitter-frontend/issues/91 .

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (this needs a follow up PR)

### Changes Made

Replaces using Redux global state step  with useParams hook, to pass relevant step of wizard. Steps are managed via routes.

### Testing

Testing is refactored to handle changed routes, both unit tests and integration tests with Cypress. There is no  additional new tests.

- [x] Unit Tests
- [x] Integration Tests
- [ ] Tests do not apply
- [ ] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ const App = (): React$Element<typeof React.Fragment> => {
       <CssBaseline />
       <NavigationMenu />
       <Switch>
-        <Route path="/newdraft">
+        <Route path="/newdraft/:step">
```
Comment (blankdots) [inline src/App.js:69]: I think just having the step might not be enough. is it possible to make it relative to also the folder id ?

Comment (lilachic) [reply src/App.js:69]: So first folderID then step in the route maybe. But where is folderID used here in wizard creating submission? And how to handle non existing folderID, when the first two steps -1 and 0 are in question, keeping folderID -1? 
folderID is kept in global state, submissionFolderSlice. And in route /home/drafts or /home/published .
Maybe not in this PR?

Comment (blankdots) [reply src/App.js:69]: i assume the non-existing folderID would only be the case when `/newdraft` is first accessed and when a request is made to the folder API that is when we know it and we can use it in the URL (and included it with the previous and next links).

i was asking if possible, sure part of another PR is ok, I was thinking of what that would imply and if the changes in this PR would need to be redone. As such a feature might be necessary when we want to link folders from `/home/drafts` to be edited in the wizard. 

Does not have to be the only solution, however evaluating our options.

Comment (lilachic) [reply src/App.js:69]: One option with this task is to do it, is to keep steps in Redux state and keep route clean. So no steps in route and try to handle browsers history other wise.

Comment (saulipurhonen) [reply src/App.js]: Id's suggest on navigating with query parameters. Example URL with params:
`/newdraft?step=1&folderid=123123`

File: src/components/NewDraftWizard/WizardComponents/WizardStepper.js [hunk 2]
```diff
@@ -149,21 +151,21 @@ const WizardStepper = ({ createFolderFormRef }: { createFolderFormRef?: CreateFo
         disableElevation
         color="primary"
         variant="outlined"
-        disabled={wizardStep < 1}
+        disabled={urlStep < 1}
```
Comment (saulipurhonen) [inline src/components/NewDraftWizard/WizardComponents/WizardStepper.js:154]: I think `wizardStep` corresponds better to what this variable stands for.

File: src/App.js (hunk 1 above)
Comment (saulipurhonen) [reply src/App.js]: Also, check and handler for `folderId` in query params would fix current problem where user refreshes the browser.

File: src/views/NewDraftWizard.js [hunk 3]
```diff
@@ -50,7 +52,7 @@ const getStepContent = (wizardStep: number, createFolderFormRef: CreateFolderFor
     case 2:
       return <WizardShowSummaryStep />
     default:
-      throw new Error("Unknown step")
+      throw new Error("Unknown step") //THIS ERROR IS NOT CATCHED
```
Comment (blankdots) [inline src/views/NewDraftWizard.js:55]: could this be solved as part of this PR, maybe 404 would make sense with the meaning: "a step that does not exist is being accessed."

File: src/components/NewDraftWizard/WizardSteps/WizardFrontpageStep.js [hunk 4]
```diff
@@ -46,7 +44,8 @@ const NewObjectTooltip = withStyles(theme => ({
  */
 const WizardFrontpageStep = (): React$Element<typeof Grid> => {
   const classes = useStyles()
-  const dispatch = useDispatch()
+  //const dispatch = useDispatch()
```
Comment (blankdots) [inline src/components/NewDraftWizard/WizardSteps/WizardFrontpageStep.js:47]: forgotten comment ?

**Review by blankdots — CHANGES_REQUESTED** [review]
other than these small comments look good.

File: src/views/NewDraftWizard.js (hunk 3 above)
Comment (blankdots) [reply src/views/NewDraftWizard.js:55]: we can redirect to `Error404` page

Comment (lilachic) [reply src/views/NewDraftWizard.js:55]: Maybe, I made a separate issue about this: https://github.com/CSCfi/metadata-submitter-frontend/issues/220 .

Comment (blankdots) [reply src/views/NewDraftWizard.js:55]: i would like this part of this PR, `401` is for `Unauthorized` which does not make sense as the user is not trying to access something (s)he does not have permissions for but something that does not exist. So 404 makes more sense.

also don't leave comments in the code, if something is not finished put in issues, leave the code clean.

File: src/components/NewDraftWizard/WizardSteps/WizardFrontpageStep.js (hunk 4 above)
Comment (saulipurhonen) [reply src/components/NewDraftWizard/WizardSteps/WizardFrontpageStep.js]: Comment deleted

File: src/views/NewDraftWizard.js (hunk 3 above)
Comment (saulipurhonen) [reply src/views/NewDraftWizard.js]: Step now defaults to 404

**Review by blankdots — APPROVED** [review]

**Review by hannyle — APPROVED** [review]
Looks great to me!

**Pull request merged**